src/reducers/starWarsReducer.js

```diff
@@ -1,13 +1,39 @@
-import /* we need our action types here*/ "../actions";
+/* we need our action types here*/
+import {
+  FETCH_CHARACTERS_START,
+  FETCH_CHARACTERS_SUCCESS,
+  FETCH_CHARACTERS_FAILURE
+} from "../actions";
+
 const initialState = {
-  characters: []
+  characters: [],
   // Array characters, Boolean fetching, null error.
+  error: "",
+  isFetching: false
 };
 export const charsReducer = (state = initialState, action) => {
   switch (action.type) {
     // Fill me in with the important reducers
     // action types should be FETCHING, SUCCESS and FAILURE
     // your switch statement should handle all of these cases.
+    case FETCH_CHARACTERS_START:
+      return {
+        ...state,
+        isFetching: true,
+        error: ""
+      };
+    case FETCH_CHARACTERS_SUCCESS:
+      return {
+        ...state,
+        characters: action.payload,
+        isFetching: false,
+        error: ""
+      };
+    case FETCH_CHARACTERS_FAILURE:
+      return {
+        ...state,
+        error: action.payload
```

Comment:
All your reducers are perfect, but don't forget to set `isFetching` to false here. Or else it will display the spinner and an error which could lead to some confusing user experience.